bin/cron/admin_stats

```diff
@@ -4,47 +4,99 @@
 #
 require File.expand_path('../../../pegasus/src/env', __FILE__)
 require src_dir 'database'
-require 'cdo/hip_chat'
 require 'cdo/properties'
 
-DASHBOARD_REPORTING_DB_READONLY = sequel_connect(CDO.dashboard_reporting_db_reader, CDO.dashboard_reporting_db_reader)
-PEGASUS_REPORTING_DB_READONLY = sequel_connect(CDO.pegasus_reporting_db_reader, CDO.pegasus_reporting_db_reader)
+DASHBOARD_REPORTING_DB_READONLY = sequel_connect(
+  CDO.dashboard_reporting_db_reader, CDO.dashboard_reporting_db_reader)
+SEVEN_DAYS = 7 * 24 * 60 * 60
+THIRTY_DAYS = 30 * 24 * 60 * 60
+
+# baseline: The set of user objects whose count is used in the denominator of the fraction.
+# Returns a hash, mapping demographics (all, gender, locale) to the ratio of users in the baseline
+#   who signed-in thirty or more days after account creation to the users in the baseline.
+def build_demographic_breakdowns(baseline)
+  comparison = baseline.where("current_sign_in_at - created_at > ?", %q{interval '30 days'})
+  students = comparison.count.to_f / [1, baseline.count].max.to_f
+  female_students = comparison.where(gender: 'f').count.to_f /
+    [1, baseline.where(gender: 'f').count].max.to_f
+  male_students = comparison.where(gender: 'm').count.to_f /
+    [1, baseline.where(gender: 'm').count].max.to_f
+  us_students = comparison.where(locale: 'en-us').count.to_f /
```

Comment:
using locale here might not be very accurate -- this is based on what they selected (or we chose, based on their browser settings as passed through request headers) on registration. It is not changed when changing language after registering and the default is en-us